### PR TITLE
leaf: Add 1.77.0

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **leaf/1.77.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/leaf/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        self.copy(pattern="LICENSE_1_0.txt", src=self._source_subfolder, dst="licenses")
+        self.copy("*", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "leaf"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "leaf"
+        self.cpp_info.names["cmake_find_package"] = "Boost"
+        self.cpp_info.names["cmake_find_package_multi"] = "Boost"
+        self.cpp_info.components["leaflib"].names["cmake_find_package"] = "leaf"
+        self.cpp_info.components["leaflib"].names["cmake_find_package_multi"] = "leaf"
```
Comment (SpaceIm) [inline recipes/leaf/all/conanfile.py:41]: targets are  not exported: https://github.com/boostorg/leaf/blob/develop/CMakeLists.txt

Comment (intelligide) [reply recipes/leaf/all/conanfile.py:41]: Leaf is part of Boost and is exported with the pattern `Boost::<lib>` like here https://github.com/boostorg/leaf/blob/develop/CMakeLists.txt#L11.  I just put this to ensure compatibility.

Comment (SpaceIm) [reply recipes/leaf/all/conanfile.py:41]: I see, it's just that leaf library alone doesn't export its target, there is no official config file.

Comment (madebr) [reply recipes/leaf/all/conanfile.py:41]: Open a PR with the following (untested) code, get it accepted and we'll happily add the targets.
```cmake
include(GNUInstallDirs)
target_include_directories(boost_leaf PUBLIC
    $<BUILD_INTERFACE:include>
    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDE_DIR}>
)
set_target_properties(boost_leaf PROPERTIES EXPORT_NAME leaf)
option(LEAF_INSTALL "Install leaf" OFF)
if(LEAF_INSTALL)
    install(TARGETS boost_leaf EXPORT leafTargets)
    install(DIRECTORY include/ DESTINATION "${CMAKE_INSTALL_INCLUDE_DIR}")
    install(EXPORT leafTargets DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/leaf/leafConfig.cmake" NAMESPACE "Boost::")
endif()
```

File: recipes/leaf/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (madebr) [inline recipes/leaf/all/test_package/conanfile.py:15]:
```suggestion
        if not tools.cross_building(self):
```

File: recipes/leaf/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+from conans import tools, ConanFile
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class BoostLeafConan(ConanFile):
+    name = "leaf"
+    description = "Lightweight Error Augmentation Framework"
+    topics = ("conan", "leaf", "boost", "error")
```
Comment (madebr) [inline recipes/leaf/all/conanfile.py:10]:
```suggestion
    topics = ("leaf", "boost", "error")
```

File: recipes/leaf/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/leaf/all/conanfile.py:41]: if it's a part of boost, is it installed with regular boost package already?

Comment (intelligide) [reply recipes/leaf/all/conanfile.py:41]: Its part of boost since 1.75. But it isn't installed with the regular boost package.

**Review by SSE4 — CHANGES_REQUESTED** [review]
please apply suggestion(s) from @madebr and we're good to go

**Pull request closed without merging**